datasets/ronec/ronec.py

```diff
@@ -0,0 +1,212 @@
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        urls_to_download = {"train": _URL + _TRAINING_FILE, "dev": _URL + _DEV_FILE, "test": _URL + _TEST_FILE}
+
+        downloaded_files = dl_manager.download(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["train"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["test"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": downloaded_files["dev"]},
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            guid = 0
+            sentence = []
+            ronec_class = []
+            start = []
+            end = []
+            sent = ""
+            has_started = False
+            for line in f:
+                if "#" in line or line == "\n":
+                    if sent:
+                        sentence.append(sent)
+                        yield guid, {
+                            "id": str(guid),
+                            "sentence": sentence,
+                            "start": start,
+                            "end": end,
+                            "ronec_class": ronec_class,
+                        }
+                        guid += 1
+                        sentence = []
+                        start = []
+                        end = []
+                        ronec_class = []
+                        sent = ""
+                else:
+                    # ronec words are tab separated
+                    line = line.replace("\n", "")
+                    splits = line.split("\t")
+                    if splits[9] == "SpaceAfter=No":
+                        sent += splits[1]
+                    else:
+                        sent += splits[1] + " "
+
+                    if splits[10].startswith("O") and not has_started:
+                        continue
+                    elif splits[10].startswith("B-"):
+                        if splits[9] == "SpaceAfter=No":
+                            begin = len(sent) - len(splits[1])
+                            last = len(sent)
+                        else:
+                            begin = len(sent) - len(splits[1]) - 1
+                            last = len(sent) - 1
+                        label = splits[10][2:]
```

Comment:
Why do you remove the `B-`, `I-` and `O-` ?
We should keep them to follow the conll IOB format as in [conll2003](https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py#L168-L178) for example